packages/api/src/external/carequality/organization.ts

```diff
@@ -1,66 +1,97 @@
-  const cqOrgDetailsString = Config.getCQOrgDetails();
-  if (!cq) return;
-  const cqOrgDetails: CQOrgDetails = cqOrgDetailsString
-    ? JSON.parse(cqOrgDetailsString)
-    : undefined;
-  if (!cqOrgDetails) {
-    const msg = "No CQ Organization details found. Skipping...";
-    console.log(msg);
-    throw new Error(msg);
+  static fromDetails(orgDetails: CQOrgDetails): CQOrganization {
+    const organization = new CQOrganization(
+      orgDetails.name,
+      orgDetails.oid,
+      orgDetails.addressLine1,
+      orgDetails.city,
+      orgDetails.state,
+      orgDetails.postalCode,
+      orgDetails.lat,
+      orgDetails.lon,
+      orgDetails.contactName,
+      orgDetails.phone,
+      orgDetails.email,
+      orgDetails.role,
+      orgDetails.hostOrgOID
+    );
+
+    this.addUrls(organization);
+    return organization;
   }
 
-  const cqOrg = buildOrganizationFromTemplate(cqOrgDetails);
-  try {
-    const org = await cq.listOrganizations({ count: 1, oid: cqOrgDetails.oid });
-    if (org.length > 0) {
-      await updateCQOrganization(cqOrg, cqOrgDetails.oid);
-      return;
-    }
-  } catch (error) {
-    console.log(`Failed to check if organization exists in the CQ Directory. Cause: ${error}`);
-    throw error;
+  static addUrls(organization: CQOrganization) {
+    organization.urlXCPD = CQOrganization.urls.urlXCPD;
+    organization.urlDQ = CQOrganization.urls.urlDQ;
+    organization.urlDR = CQOrganization.urls.urlDR;
   }
 
-  try {
-    console.log(`Registering organization in the CQ Directory...`);
-    await cq.registerOrganization(cqOrg);
-  } catch (error) {
-    console.log(`Failed to register organization in the CQ Directory. Cause: ${error}`);
-    throw error;
+  public getDetails(): CQOrgDetails {
+    return {
+      name: this.name,
+      oid: this.oid,
+      addressLine1: this.addressLine1,
+      city: this.city,
+      state: this.state,
+      postalCode: this.postalCode,
+      lat: this.lat,
+      lon: this.lon,
+      contactName: this.contactName,
+      phone: this.phone,
+      email: this.email,
+      role: this.role,
+      hostOrgOID: this.hostOrgOID,
+    };
+  }
+
+  public getUrls(): CQOrgUrls {
+    return CQOrganization.urls;
   }
-}
 
-async function updateCQOrganization(cqOrg: string, oid: string): Promise<void> {
-  console.log(`Updating org in the CQ Directory...`);
-  if (!cq) return;
-  try {
-    await cq.updateOrganization(cqOrg, oid);
-  } catch (error) {
-    console.log(`Failed to update organization in the CQ Directory. Cause: ${error}`);
-    throw error;
+  public getDetailsAndUrls(): CQOrgUrls & CQOrgDetails {
+    return {
+      ...this.getDetails(),
+      ...this.getUrls(),
+    };
+  }
+
+  public buildXmlStringFromTemplate(): string {
+    return buildXmlStringFromTemplate(this.getDetailsAndUrls());
+  }
+
+  public toXmlString(): string {
+    if (!this.xmlString) {
+      this.xmlString = this.buildXmlStringFromTemplate();
+    }
+    return this.xmlString;
```

Comment:
This is good IFF the object is immutable. Otherwise one can call `toXmlString`, update a property, and call `toXmlString` only to get an outdated version of the object.

Reply:
Very good point! Fixed